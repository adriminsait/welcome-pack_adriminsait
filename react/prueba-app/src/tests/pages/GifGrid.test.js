import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom'

import { GifGrid } from '../../pages/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');

describe('pruebas <GifGrid>', () => {

    const category = 'One Punch';

    test('probando que se ve', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })

        const wrapper = shallow(<GifGrid category={category}/>);

        expect(wrapper).toMatchSnapshot();
    });

    test('debe mostrar items cuando cargan imagenes useFetchGifs', () => {

        const gifs = [
        {
            id: 'abc',
            url: 'algo',
            title: 'otra cosa'
        },
        {
            id: 'def',
            url: 'algo',
            title: 'otra cosa'
        }
        ];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        })

        const wrapper = shallow(<GifGrid category={category}/>);

        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    });
    
});
