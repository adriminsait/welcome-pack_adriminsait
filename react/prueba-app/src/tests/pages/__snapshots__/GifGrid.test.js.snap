// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas <GifGrid> probando que se ve 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    One Punch
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
